app/Migrations/Version20220726103404.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+/**
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20220726103404 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->addSql('CREATE TEMPORARY TABLE __temp__sdk_setting AS SELECT id, path, "values", strategy, type, is_project, has_initialization, initialization_description, initializer FROM sdk_setting');
+        $this->addSql('DROP TABLE sdk_setting');
+        $this->addSql('CREATE TABLE sdk_setting (id INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, path VARCHAR(255) NOT NULL, "values" CLOB DEFAULT NULL --(DC2Type:json)
+        , strategy VARCHAR(255) NOT NULL, type VARCHAR(255) NOT NULL, has_initialization BOOLEAN NOT NULL, initialization_description VARCHAR(255) DEFAULT NULL, initializer VARCHAR(255) DEFAULT NULL, setting_type CHECK(setting_type IN (\'sdk\',\'local\',\'shared\')))');
+        $this->addSql('INSERT INTO sdk_setting (id, path, "values", strategy, type, setting_type, has_initialization, initialization_description, initializer) SELECT id, path, "values", strategy, type, (CASE WHEN is_project = 1 THEN "local" ELSE "sdk" END) AS setting_type , has_initialization, initialization_description, initializer FROM __temp__sdk_setting');
+        $this->addSql('DROP TABLE __temp__sdk_setting');
+        $this->addSql('CREATE UNIQUE INDEX UNIQ_8463B8E1B548B0F ON sdk_setting (path)');
```

Comment:
If i got it right you've just added the `setting_type` field and populate it. This solution is ok because we have the small table. Usually the nullable field is added -> populate it  -> remove nullable.

Reply:
we can't create one field and remote it then. it's SQLite